### ISSUE TITLE
Turn .plot() method in accessor so other pandas plotting methods are also available

### ISSUE THREAD

**Issue by jorisvandenbossche** [geopandas/geopandas#766]
Currently `GeoDataFrame.plot()` function overrides the pandas plot method, which means you cannot quickly plot your non-geometry columns. 

The idea would be to follow the pattern from pandas where the plot() method is also an accessor like .plot.scatter() so those become available, while the default `.plot()` would still override with the geometrical functionality.

**Comment by pulkitmaloo** [comment]
I'd like to take up this issue. Can someone help me how should I proceed?

**Comment by benabernathy** [comment]
Sorry for the stream of consciousness here. I'm kind of working things out as I type. I looked a little closer at how Pandas accomplishes this. It has a class called `FramePlotMethod` that can be called either as a method or used as an accessor where the other plotting methods can be called (e.g. `df.plot.line()`. The class is added as a property to `DataFrame` through the use of another Pandas class called `CachedAccessor`. 

It appears GeoPandas does things a little bit differently. GeoPandas has a set of plotting methods defined in `plotting.py` as well as a `plot()` method on the `GeoDataFrame` class. 

It seems to me the best long term approach is to make a plotting class in GeoPandas similar to `FramePlotMethod` (e.g. `GeoPlotMethod`). Then add it as a property by wrapping it in some sort of accessor. Maybe we can take advantage of Pandas's `CachedAccessor` here?  

However, I am not sure of the impact of moving those plotting methods in `plotting.py` into a class. That seems like it would be a breaking change in the API. We should be sensitive to that. 

An alternative is we create the class, but not migrate those functions into the new class. The class could proxy those functions, then over time migrate them to the class during a major version change.

***TL;DR Version***
1. Create a new class called `GeoPlotMethods`.
2. `GeoPlotMethods` should proxy calls to original methods in `plotting.py`.
3. Add `GeoPlotMethods` to `GeoDataFrame` as a property using an accessor class such as `CachedAccessor`.
4. Make `GeoPlotMethods` callable and migrate the `GeoDataFrame.plot()` method from the `GeoDataFrame` class to `GeoPlotMethods`. 
5. Find a way to proxy the call to a Pandas `DataFrame.plot`. This could possibly be done by having our new `GeoPlotMethods` class extend `FramePlotMethod`. This would allow for calls like `gdf.plot.line()`, `gdf.plot()` as well as future plotting techniques.

Thoughts? Criticisms?

**Comment by martinfleis** [comment]
xref #354

**Comment by martinfleis** [comment]
Closed by #1465
